upup/pkg/fi/cloudup/awstasks/natgateway.go

```diff
@@ -262,8 +311,15 @@ func (_ *NatGateway) RenderAWS(t *awsup.AWSAPITarget, a, e, changes *NatGateway)
 	tags["AssociatedNatgateway"] = *id
 	err := t.AddAWSTags(*e.Subnet.ID, tags)
 	if err != nil {
-		return fmt.Errorf("Unable to tag subnet %v", err)
+		return fmt.Errorf("unable to tag subnet %v", err)
+	}
+
+	// If this is a shared NGW, we need to leave some additional tags so we don't delete existing resources
+	if *e.Shared == true {
+		glog.V(2).Infof("tagging route table %s to track shared NGW", *e.AssociatedRouteTable.ID)
+		err = t.AddAWSTags(*e.AssociatedRouteTable.ID, tags)
```

Comment:
i.e. a special tag indicates that we are using the nat gateway, but that we don't own the nat gateway